assets/app/views/project.html

```diff
@@ -17,7 +17,24 @@
         <div class="row">
           <div class="col-md-12">
             <div class="tile">
-              <h2 class="service osc-object" ng-init="numPorts = service.spec.ports.length" kind="Service" resource="service">{{serviceId}}
+
+              <!-- Routes -->
+              <div ng-if="displayRouteByService[serviceId]" ng-init="route = displayRouteByService[serviceId]">
```

Comment:
can you put the `ng-if` directly on the h2? try dropping the `ng-init`s, using the full `displayRouteByService[serviceId]` reference, and see if that helps osc-object behave better